File: app/src/main/java/com/laushkina/anastasia/psychosophy/view/test/TestViewModel.java
package com.laushkina.anastasia.psychosophy.view.test;

import android.databinding.BaseObservable;
import android.databinding.Bindable;

import com.laushkina.anastasia.psychosophy.BR;

public class TestViewModel extends BaseObservable {
    private Boolean isNextEnabled;
    private Boolean isPrevEnabled;
    private int progress;
    private CharSequence nextButtonText;
    private CharSequence prevButtonText;

    TestViewModel(CharSequence nextButtonText){
        this();
        this.nextButtonText = nextButtonText;
    }

    private TestViewModel(){
        isNextEnabled = false;
        isPrevEnabled = false;
        progress = 0;
    }

    @Bindable
    public boolean isNextEnabled() {
        return isNextEnabled;
    }

    public void setNextEnabled(boolean nextEnabled) {
        isNextEnabled = nextEnabled;
        notifyPropertyChanged(BR.nextEnabled);
    }

    @Bindable
    public int getProgress() {
        return progress;
    }

    public void setProgress(int progress) {
        this.progress = progress;
        notifyPropertyChanged(BR.progress);
    }

    @Bindable
    public CharSequence getNextButtonText() {
        return nextButtonText;
    }

    public void setNextButtonText(CharSequence nextButtonText) {
        this.nextButtonText = nextButtonText;
        notifyPropertyChanged(BR.nextButtonText);
    }

    @Bindable
    public CharSequence getPrevButtonText() {
        return prevButtonText;
    }

    public void setPrevButtonText(CharSequence prevButtonText) {
        this.prevButtonText = prevButtonText;
        notifyPropertyChanged(BR.prevButtonText);
    }

    @Bindable
    public Boolean getPrevEnabled() {
        return isPrevEnabled;
    }

    public void setPrevEnabled(Boolean prevEnabled) {
        isPrevEnabled = prevEnabled;
        notifyPropertyChanged(BR.prevEnabled);
    }
}

File: app/src/main/java/com/laushkina/anastasia/psychosophy/view/test/TestViewModel.kt
package com.laushkina.anastasia.psychosophy.view.test

import android.databinding.BaseObservable
import android.databinding.Bindable
import com.laushkina.anastasia.psychosophy.BR

class TestViewModel(private var isNextEnabled: Boolean,
                    private var isPrevEnabled: Boolean,
                    private var progress: Int): BaseObservable() {

    private lateinit var nextButtonText: CharSequence
    private var prevButtonText: CharSequence = ""

    constructor(): this(false, false, 0)

    @Bindable
    fun isNextEnabled(): Boolean {
        return isNextEnabled
    }

    fun setNextEnabled(nextEnabled: Boolean) {
        this.isNextEnabled = nextEnabled
        notifyPropertyChanged(BR.nextEnabled)
    }

    @Bindable
    fun getProgress(): Int {
        return progress
    }

    fun setProgress(progress: Int) {
        this.progress = progress
        notifyPropertyChanged(BR.progress)
    }

    @Bindable
    fun getNextButtonText(): CharSequence {
        return nextButtonText
    }

    fun setNextButtonText(nextButtonText: CharSequence) {
        this.nextButtonText = nextButtonText
        notifyPropertyChanged(BR.nextButtonText)
    }

    @Bindable
    fun getPrevButtonText(): CharSequence {
        return prevButtonText
    }

    fun setPrevButtonText(prevButtonText: CharSequence) {
        this.prevButtonText = prevButtonText
        notifyPropertyChanged(BR.prevButtonText)
    }

    @Bindable
    fun getPrevEnabled(): Boolean? {
        return isPrevEnabled
    }

    fun setPrevEnabled(prevEnabled: Boolean) {
        this.isPrevEnabled = prevEnabled
        notifyPropertyChanged(BR.prevEnabled)
    }
}